onnxruntime/test/featurizers_ops/rolling_window_transformer_test.cc

```diff
@@ -25,79 +26,111 @@ std::vector<uint8_t> GetStream(EstimatorT& estimator, const std::vector<InputTyp
 
   auto stream = GetStream(estimator, training_batch);
   auto dim = static_cast<int64_t>(stream.size());
-  OpTester test("RollingWindowTransformer", 1, onnxruntime::kMSFeaturizersDomain);
+  OpTester test("AnalyticalRollingWindowTransformer", 1, onnxruntime::kMSFeaturizersDomain);
   test.AddInput<uint8_t>("State", {dim}, stream);
   test.AddInput<std::string>("Grains", {4, 1}, {"one", "two", "one", "two"});
-  test.AddInput<int32_t>("Target", {4}, {1, 1, 2, 2});
-  test.AddOutput<double>("Output", {4, 2}, {NS::Traits<double>::CreateNullValue(),
-                                            NS::Traits<double>::CreateNullValue(),
-                                            NS::Traits<double>::CreateNullValue(),
-                                            NS::Traits<double>::CreateNullValue(),
-                                            NS::Traits<double>::CreateNullValue(),
-                                            1.0,
-                                            NS::Traits<double>::CreateNullValue(),
-                                            1.0});
+  test.AddInput<double>("Target", {4}, {1, 1, 2, 2});
+
+  test.AddOutput<double>("Output", {4, 1, 2}, {NS::Traits<double>::CreateNullValue(),
+                                               NS::Traits<double>::CreateNullValue(),
+                                               NS::Traits<double>::CreateNullValue(),
+                                               NS::Traits<double>::CreateNullValue(),
+                                               NS::Traits<double>::CreateNullValue(),
+                                               1.0,
+                                               NS::Traits<double>::CreateNullValue(),
+                                               1.0});
+
+  test.Run();
+}
+
+TEST(FeaturizersTests, SimpleRollingWindow_Transformer_Grained_Min_1_grain_window_size_1_horizon_1) {
+  using EstimatorT = NS::Featurizers::GrainedSimpleRollingWindowEstimator<double>;
+
+  EstimatorT      estimator(NS::CreateTestAnnotationMapsPtr(1), NS::Featurizers::SimpleRollingWindowCalculation::Min, 1, 1);
+  using GrainType = std::vector<std::string>;
+  using GrainedInputType = EstimatorT::InputType;
+  const GrainType grainOne({"one"});
+  const double value1(static_cast<double>(1));
+  const GrainedInputType tup1(grainOne, value1);
+  const std::vector<std::tuple<std::vector<std::string> const &, double const &>> training_batch = {tup1};
+
+  auto stream = GetStream(estimator, training_batch);
+  auto dim = static_cast<int64_t>(stream.size());
+  OpTester test("SimpleRollingWindowTransformer", 1, onnxruntime::kMSFeaturizersDomain);
+  test.AddInput<uint8_t>("State", {dim}, stream);
+  test.AddInput<std::string>("Grains", {3, 1}, {"one", "one", "one"});
+  test.AddInput<double>("Target", {3}, {1, 2, 3});
+
+  test.AddOutput<double>("Output", {3, 1, 1}, {NS::Traits<double>::CreateNullValue(), 1, 2});
```

Comment:
Nit: Adding vertical whitespace would make this code easier to read - especially between using statements, variable declarations, and statements.